tests: bluetooth: RX overflow in lt_tx_real_no_encode

**Describe the bug**
PR #54119 describes a bug, but a proper fix requires more than is described in the PR.
when compiling tests/bluetooth/controller/ctrl_feature_exchange/bluetooth.controller.ctrl_feature_exchange.test without -ffreestanding (i.e., with memcpy warnings enabled), the following error pops up:

.../tests/bluetooth/controller/common/src/helper_util.c: In function ‘lt_tx_real_no_encode’: .../tests/bluetooth/controller/common/src/helper_util.c:412:9: error: ‘memcpy’ writing 39 bytes into a region of size 32 overflows the destination [-Werror=stringop-overflow=]
412 | memcpy((struct pdu_data *)rx->pdu, pdu, sizeof(struct pdu_data));
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.../tests/bluetooth/controller/common/src/helper_util.c:410:14: note: at offset 32 into destination object of size 64 allocated by ‘malloc’
410 | rx = malloc(PDU_RX_NODE_SIZE);
| ^~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors



Please also mention any information which could help others to understand
the problem you're facing:
 Most likely there is an issue with the define of PDU_RX_NODE_SIZE in helper_util.c  PDU_DATA_SIZE should use sizeof(struct pdu_data) 

**To Reproduce**
Steps to reproduce the behavior:
see above

**Expected behavior**
no compiler error should happen

**Impact**
low priority

**Logs and console output**
N/A

**Environment (please complete the following information):**
 N/A

**Additional context**
N/A

--- a/tests/bluetooth/controller/common/src/helper_util.c
+++ b/tests/bluetooth/controller/common/src/helper_util.c
@@ -57,7 +57,7 @@ static uint32_t no_of_ctx_buffers_at_test_setup;
 #define PDU_DC_LL_HEADER_SIZE (offsetof(struct pdu_data, lldata))
 #define NODE_RX_HEADER_SIZE (offsetof(struct node_rx_pdu, pdu))
 #define NODE_RX_STRUCT_OVERHEAD (NODE_RX_HEADER_SIZE)
-#define PDU_DATA_SIZE (PDU_DC_LL_HEADER_SIZE + LL_LENGTH_OCTETS_RX_MAX)
+#define PDU_DATA_SIZE sizeof(struct pdu_data)
 #define PDU_RX_NODE_SIZE WB_UP(NODE_RX_STRUCT_OVERHEAD + PDU_DATA_SIZE)
 
 helper_pdu_encode_func_t *const helper_pdu_encode[] = {

